### PR TITLE
FAT-388563 Cancel invoice created in current FY and paid against previous FY (Thunderjet) (TaaS)

### PR DESCRIPTION
**Pull request by dnovikau**
[C388563](https://foliotest.testrail.io/index.php?/cases/view/388563)
***
[Allure report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4841/allure)
***

### CODE REVIEW

File: cypress/e2e/invoices/cancel-invoice-created-in-current-fy-and-paid-against-previous-fy.cy.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+import { INVOICE_STATUSES } from '../../support/constants';
+
+describe('Invoices', () => {
+  const date = new Date();
+  const code = CodeTools(4);
+  const fiscalYears = {
+    first: {
+      ...FiscalYears.getDefaultFiscalYear(),
+      code: `${code}${StringTools.randomTwoDigitNumber()}01`,
+      periodStart: new Date(date.getFullYear(), date.getMonth() - 1, date.getDate()),
```
Comment (IhorBohdan) [inline cypress/e2e/invoices/cancel-invoice-created-in-current-fy-and-paid-against-previous-fy.cy.js:21]: There are already out-of-the-box methods in the DateTools. You can use them.

File: cypress/support/fragments/invoices/invoices.js [hunk 2]
```diff
@@ -120,9 +120,10 @@ export default {
       })
       .then(({ body }) => body);
   },
-  createInvoiceViaApi({ vendorId, accountingCode, exportToAccounting }) {
+  createInvoiceViaApi({ vendorId, accountingCode, fiscalYearId, exportToAccounting }) {
```
Comment (IhorBohdan) [inline cypress/support/fragments/invoices/invoices.js:123]: It's not a necessary element. You don't need to add it. You can create an invoice without it.

File: cypress/support/fragments/invoices/modal/cancelInvoiceModal.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+export default {
+  verifyModalView() {
+    cy.expect([
+      cancelInvoiceConfirmationModal.has({
+        header: 'Cancel invoice',
+      }),
+      cancelInvoiceConfirmationModal.has({
+        message: including(
+          'Are you sure you want to cancel this invoice? Any related transactions against funds will be voided.',
```
Comment (IhorBohdan) [inline cypress/support/fragments/invoices/modal/cancelInvoiceModal.js:20]: You can remove message to const.

**Review by IhorBohdan — APPROVED** [review]

File: cypress/support/fragments/invoices/invoices.js (hunk 2 above)
Comment (dnovikau) [reply cypress/support/fragments/invoices/invoices.js:123]: I need this for test purposes. Test precondition contain requirement to create Invoice with specified FY

Comment (IhorBohdan) [reply cypress/support/fragments/invoices/invoices.js:123]: So it's only in your test. Not all tests are like this. Why change the whole fragment?

File: cypress/e2e/invoices/cancel-invoice-created-in-current-fy-and-paid-against-previous-fy.cy.js (hunk 1 above)
Comment (dnovikau) [reply cypress/e2e/invoices/cancel-invoice-created-in-current-fy-and-paid-against-previous-fy.cy.js]: Fixed

File: cypress/support/fragments/invoices/modal/cancelInvoiceModal.js (hunk 3 above)
Comment (dnovikau) [reply cypress/support/fragments/invoices/modal/cancelInvoiceModal.js]: Fixed

File: cypress/support/fragments/invoices/invoices.js (hunk 2 above)
Comment (dnovikau) [reply cypress/support/fragments/invoices/invoices.js:123]: Other tests will not be affected, if 'fiscalYearId' is not provided as an argument then request body to create invoice will not contain 'fiscalYearId' property. Previous behavior of the function will not change.

**Review by Scorpibear — APPROVED** [review]

**Pull request merged**